aspnetcore/performance/performance-best-practices.md

```diff
@@ -0,0 +1,129 @@
+
+Recommendations:
+
+* **Do** consider caching large objects that are frequently used so that they don't need to be reallocated each time they're needed.
+* **Do** pool buffers by using an [`ArrayPool<T>`](/dotnet/api/system.buffers.arraypool-1) to store large arrays.
+* **Do not** allocate many, short-lived large objects on [hot code paths](#hot).
+
+Memory issues like the preceding can be diagnosed by reviewing garbage collection (GC) stats in [PerfView](https://github.com/Microsoft/perfview) and examining:
+
+* Garbage collection pause time.
+* What percentage of the processor time is spent in garbage collection.
+* How many garbage collections are generation 0, 1, and 2.
+
+For more information, see [Garbage Collection and Performance](/dotnet/standard/garbage-collection/performance).
+
+## Optimize Data Access
+
+<!-- TODO review by EF folks -->
+
+Interactions with a data store or other remote services are often the slowest parts of an ASP.NET Core app. It's important to make sure that data is read and written efficiently. In addition to ensuring data access is asynchronous, some best practices include:
+
+* **Do** consider caching frequently accessed data retrieved from a database or remote service if it is acceptable for the data to be slightly out-of-date. Depending on the scenario, you might use a [MemoryCache](xref:performance/caching/memory) or a [DistributedCache](xref:performance/caching/distributed). For more information, see [Cache responses in ASP.NET Core](xref:performance/caching/index).
+* Prefer *chunky* over *chatty* database interactions. A *chatty* interaction requires many calls to the data store. A *chunky* interaction requires fewer network calls. The goal is to retrieve all the data that will be needed in a single call rather than  several calls.
+* **Do** use [no-tracking queries](/ef/core/querying/tracking) in Entity Framework Core when accessing read-only data.
+* **Do** filter and aggregate LINQ queries (with `.Where`, `.Select`, or `.Sum` statements, for example) so that the filtering is done by the database.
+* **Do not** retrieve more data than is necessary. Write queries to return just the data that is necessary for the current HTTP request.
+
+Query issues can be detected by reviewing time spent accessing data with [Application Insights](/azure/application-insights/app-insights-overview) or with profiling tools. Most databases also make statistics available concerning frequently executed queries.
+
```

Comment:
For customers that want to extract the last bit of performance we can also point to explicitly compiled queries https://docs.microsoft.com/ef/core/what-is-new/ef-core-2.0#explicitly-compiled-queries and DbContext pooling https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-2.0#dbcontext-pooling. But we should recommend to customers to measure the actual impact these features could have in their applications, because they have limitations and compiled queries make the application code more complicated.

Reply:
Agreed. This would be a useful resource to link to.